ngx-tools/src/update-control-on-input-changes/update-control-on-input-changes.ts

```diff
@@ -2,29 +2,33 @@ import { SimpleChanges } from '@angular/core';
 import { AbstractControl } from '@angular/forms';
 
 import { inputHasChanged } from './../input-has-changed/input-has-changed';
+import { objectDeepParse } from '../utilities/object-deep-parse';
 
 
 /**
  * Helper function to determine if a specific value has changed
  *
  * @param changes - The object of changes
- * @param key - The object key in question
  * @param control - The formControl
  * @return True if the value has changed
  */
 export function updateControlOnInputChanges(
   changes: SimpleChanges,
-  key: string,
+  path: string,
   control: AbstractControl | null,
 ): boolean {
-  if (!changes || !key || !control) {
+  if (!changes || !path || !control) {
     return false;
   }
 
-  if (inputHasChanged(changes, key)) {
-    control.setValue(changes[key].currentValue);
+  const keys = path.split('.');
+  const key = keys.shift();
+  if (inputHasChanged(changes, path) && key) {
+    const newValue = objectDeepParse(changes[key].currentValue, keys);
+    control.setValue(newValue);
```

Comment:
If `objectDeepParse` doesn't find the key passed in, we'll get `undefined` back.. should we set that to the control still? What do you think?